### PR TITLE
CB-1570 - env removed from CB and wired to env service

### PR DESCRIPTION
**Pull request by attilapalfi92**
- UI still needs to be updated
- didn't delete the endpoints. easier transition for the integration tests
- Todo: fix remaining few unit tests, disable not working int tests.

### CODE REVIEW

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 1]
```diff
@@ -41,75 +41,67 @@
...
+    SimpleEnvironmentResponses list(@PathParam("workspaceId") Long workspaceId);
 
     @PUT
-    @Path("/{name}/attach")
+    @Path("/{crn}/attach")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.ATTACH_RESOURCES, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "attachResourcesToEnvironment")
-    DetailedEnvironmentV4Response attach(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
+    DetailedEnvironmentV4Response attach(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
```
Comment (bergerdenes) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: Shouldn't it be DetailedEnvironmentResponse ?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 2]
```diff
@@ -41,75 +41,67 @@
...
             @Valid EnvironmentAttachV4Request request);
 
     @PUT
-    @Path("/{name}/detach")
+    @Path("/{crn}/detach")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.DETACH_RESOURCES, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "detachResourcesFromEnvironment")
-    DetailedEnvironmentV4Response detach(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
+    DetailedEnvironmentV4Response detach(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
```
Comment (bergerdenes) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: Shouldn't it be DetailedEnvironmentResponse ?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 3]
```diff
@@ -41,75 +41,67 @@
...
-
-    @PUT
-    @Path("/{name}/register_datalake")
+    @Path("/{crn}/register_datalake")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.REGISTER_EXTERNAL_DATALAKE, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "registerExternalDatalake")
-    DetailedEnvironmentV4Response registerExternalDatalake(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
-        @Valid RegisterDatalakeV4Request request);
+    DetailedEnvironmentV4Response registerExternalDatalake(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
```
Comment (bergerdenes) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: Shouldn't it be DetailedEnvironmentResponse ?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java [hunk 4]
```diff
@@ -11,71 +11,66 @@
...
     @Override
-    public DetailedEnvironmentV4Response edit(Long workspaceId, String environmentName, @NotNull EnvironmentEditV4Request request) {
-        return environmentService.edit(workspaceId, environmentName, request);
+    public DetailedEnvironmentResponse edit(Long workspaceId, String environmentCrn, @NotNull EnvironmentEditRequest request) {
+        return environmentClientService.edit(environmentCrn, request);
     }
 
     @Override
-    public DetailedEnvironmentV4Response attach(Long workspaceId, String environmentName, @Valid EnvironmentAttachV4Request request) {
+    public DetailedEnvironmentV4Response attach(Long workspaceId, String environmentCrn, @Valid EnvironmentAttachV4Request request) {
```
Comment (bergerdenes) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java]: Shouldn't it be DetailedEnvironmentResponse ?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java [hunk 5]
```diff
@@ -11,71 +11,66 @@
...
         throw new UnsupportedOperationException("Attaching resource to an environment is not supported anymore!");
     }
 
     @Override
-    public DetailedEnvironmentV4Response detach(Long workspaceId, String environmentName, @Valid EnvironmentDetachV4Request request) {
+    public DetailedEnvironmentV4Response detach(Long workspaceId, String environmentCrn, @Valid EnvironmentDetachV4Request request) {
```
Comment (bergerdenes) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java]: Shouldn't it be DetailedEnvironmentResponse ?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java [hunk 6]
```diff
@@ -11,71 +11,66 @@
...
 
     @Override
-    public DetailedEnvironmentV4Response changeCredential(Long workspaceId, String environmentName, @Valid EnvironmentChangeCredentialV4Request request) {
-        return environmentService.changeCredential(environmentName, workspaceId, request);
-    }
-
-    @Override
-    public DetailedEnvironmentV4Response registerExternalDatalake(Long workspaceId, String environmentName, @Valid RegisterDatalakeV4Request request) {
-        return environmentService.registerExternalDatalake(environmentName, workspaceId, request);
+    public DetailedEnvironmentV4Response registerExternalDatalake(Long workspaceId, String environmentCrn, @Valid RegisterDatalakeV4Request request) {
```
Comment (bergerdenes) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java]: Shouldn't it be DetailedEnvironmentResponse ?

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java [hunk 7]
```diff
@@ -14,20 +14,20 @@
 
 import com.sequenceiq.cloudbreak.api.endpoint.v4.environment.requests.DatalakePrerequisiteV4Request;
```
Comment (bergerdenes) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: Are these imports going to be moved under Environment service as well?

Comment (attilapalfi92) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: If it should go to the Environment service, then yes/

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java (hunk 1 above)
Comment (attilapalfi92) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java (hunk 2 above)
Comment (attilapalfi92) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java (hunk 3 above)
Comment (attilapalfi92) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java]: Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java (hunk 4 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java]: Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackV4RequestValidator.java [hunk 8]
```diff
@@ -145,8 +140,11 @@ private void validateEncryptionKey(StackV4Request stackRequest, ValidationResult
                     EncryptionType valueForTypeKey = getEncryptionType(request.getTemplate());
                     return EncryptionType.CUSTOM.equals(valueForTypeKey);
                 })
-                .forEach(request -> checkEncryptionKeyValidityForInstanceGroupWhenKeysAreListable(request, stackRequest.getEnvironment().getCredentialName(),
-                        stackRequest.getPlacement().getRegion(), validationBuilder, workspaceId));
+                .forEach(request -> {
+                    DetailedEnvironmentResponse environment = environmentClientService.get(stackRequest.getEnvironmentCrn());
```
Comment (bergerdenes) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackV4RequestValidator.java]: optimize it by moving it to the first line of this method

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java (hunk 5 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java]: Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java (hunk 6 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/EnvironmentV4Controller.java]: Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.

File: core/src/main/java/com/sequenceiq/cloudbreak/repository/cluster/DatalakeResourcesRepository.java [hunk 9]
```diff
@@ -30,10 +29,14 @@
     Optional<DatalakeResources> findByDatalakeStackId(Long datalakeStackId);
 
     @CheckPermissionsByWorkspaceId
-    @Query("SELECT dr.name FROM DatalakeResources dr WHERE dr.workspace.id = :workspaceId AND dr.environment.id = :envId")
-    Set<String> findDatalakeResourcesNamesByWorkspaceAndEnvironment(@Param("workspaceId") Long workspaceId, @Param("envId") Long envId);
+    @Query("SELECT dr.name FROM DatalakeResources dr WHERE dr.workspace.id = :workspaceId AND dr.environmentCrn = :envCrn")
```
Comment (bergerdenes) [inline core/src/main/java/com/sequenceiq/cloudbreak/repository/cluster/DatalakeResourcesRepository.java]: Parameter naming is not consistent

**Comment by attilapalfi92** [comment]
Closing this because it will be merge with another change together.

**Pull request closed without merging**